test/model.user-identity.test.js

```diff
@@ -244,4 +244,57 @@ describe('UserIdentity', function() {
       done();
     });
   });
+
+  describe('emailOptional set to false', function(err) {
+    function customProfileToUser(provider, profile, options) {
+      var userInfo = {
+        username: profile.username,
+        password: 'secret',
+      };
+      if (profile.emails && profile.emails[0] && profile.emails[0].value) {
+        userInfo.email = profile.emails[0].value;
+      }
+      return userInfo;
+    }
+
+    it('creates a user when given an email address', function(done) {
+      UserIdentity.login('facebook', 'oAuth 2.0', {
+        emails: [{ value: 'abc123@fb.com' }],
+        id: 'fbi123',
+        username: 'fbu123',
+      }, {
+        accessToken: 'at1',
+        refreshToken: 'rt1',
+      }, {
+        emailOptional: false,
+        profileToUser: customProfileToUser,
+      }, function(err, user, identity, token) {
+        assert(!err, 'No error should be reported');
+        assert(user, 'User instance should be returned');
```

Comment:
You should verify the user instance returned has the correct values (ie. the correct user was returned after you logged in).